### ISSUE TITLE
'google_beta' provider not found

### ISSUE THREAD

**Issue by willkessler** [pulumi/pulumi-gcp#586]
Just trying to gain access to gcp secret in  cloudrun configuration. Getting this error:
``` 
  index.ts(66,15): error TS2304: Cannot find name 'google_beta'.
 ```
## Steps to reproduce

[This code](https://www.pulumi.com/docs/reference/pkg/gcp/cloudrun/service/#cloud-run-service-secret-environment-variables) from the examples will do it:

```
const project = gcp.organizations.getProject({});
const secret = new gcp.secretmanager.Secret("secret", {
    secretId: "secret",
    replication: {
        automatic: true,
    },
}, {
    provider: google_beta,
});
```

**Expected**: I expected to be able to use a GCP secret to pass as an environment variable into a cloud run container
**Actual**: Typescript error

Note that I have checked all the source code in the GCP repo. `google_beta` is present in every example; yet I don't find it defined in the code.

I have run `npm i` as well as `npm i @pulimi/gcp` .

I am able to create a cloud run service and even connect it to postgres, but I want to put the PG password into a GCP secret, not a Pulumi secret.

Thanks for any guidance or explanation of this behavior.

**Comment by lukehoban** [comment]
The generated examples don't always create all the resources they depend on.  In this case, the `goole_beta` resource is not defined there.  You will need to add `const google_beta = new google.Provider(...)` to construct a Provider resource for the GCP provider.  See for example https://www.pulumi.com/docs/reference/pkg/gcp/provider/.

**Comment by lukehoban** [comment]
We have talked about identifying these cases where there are variables that aren't defined in the example, and adding a note of some kind to the example/docs to make this more clear - but it is not yet clear how best to do that.

**Comment by soonmyeongyoon** [comment]
tldr; drop google_beta provider reference, it works without it. Add annotations under metadata {'run.google.apis.com/launch-stage': 'BETA'} under both metadata.annotations and template.metadata.annotations.

@willkessler Not sure if you got to the bottom of this - was following the exact same section and gave me a cryptic headache. It looks like it is not so much of an issue of google_beta provider. Feel free to drop the google_beta provider option at the end when you create the secret - it works without it.

Reference to google_beta provider looks like a documentation hangover when this was adapted from Terraform provider. Pulumi's GCP provider is already built on Terraform's google-beta provider if the package detail is correct per Pulumi's documentation.
![image](https://user-images.githubusercontent.com/3670529/134074997-26eed00e-a8ec-4450-a227-b970ceb39d31.png)

What really blocked me was referencing the secret in the actual cloudrun.Service resource definition; when I tried to refer back to the secret version, it will throw an error mentioned at https://github.com/hashicorp/terraform-provider-google/issues/9159.

Turns out GCP APIs do expect an extra annotation for their beta features, which should be added under metadata.annotations and template.metadata.annotations.

![image](https://user-images.githubusercontent.com/3670529/134075858-97d1c8a4-da41-412c-8da2-b7f21de7b12f.png)
![image](https://user-images.githubusercontent.com/3670529/134075912-1d1046cd-bfaa-4c8c-be6a-fffad878a335.png)

**Comment by willkessler** [comment]
thanks @soonmyeongyoon because of this blocker I dropped Pulumi and reverted to terraform for everything. Maybe the next proj i can return to Pulumi...
